nucleus/core/logging/src/main/java/com/sun/enterprise/server/logging/GFFileHandler.java

```diff
@@ -585,14 +573,10 @@ private void configureUniformLogFormatter(String excludeFields, boolean multiLin
     }
 
     void initializePump() {
-        if (pumpFuture != null) {
-            pumpFuture.cancel(true);
-            pumpFuture = null;
-        }
-        if (logToFile) {
-            pumpFuture = payaraExecutorService.submit(
-                () -> {
-                    while (!done.isSignalled() && logToFile) {
+        pump = new Thread() { //Not using the PayaraExecutorService here as it prevents shutdown happening quickly, see PAYARA-4118
```

Comment:
This will create and start the pump thread even if the logToFile property may be false. Admitted that it will finish quickly but nevertheless.